io/tools/convert_clouds_meshes.cpp

```diff
@@ -0,0 +1,311 @@
+  std::vector<int> pcd_files = pcl::console::parse_file_extension_argument (argc, argv, ".pcd");
+  std::vector<int> ply_files = pcl::console::parse_file_extension_argument (argc, argv, ".ply");
+  std::vector<int> stl_files = pcl::console::parse_file_extension_argument (argc, argv, ".stl");
+  std::vector<int> vtk_files = pcl::console::parse_file_extension_argument (argc, argv, ".vtk");
+  std::string parsed_output_type;
+  pcl::console::parse_argument (argc, argv, "-f", parsed_output_type);
+  pcl::console::parse_argument (argc, argv, "--format", parsed_output_type);
+
+  // Make sure that we have one input and one output file only
+  if (obj_files.size () + pcd_files.size () + ply_files.size () + stl_files.size () + vtk_files.size () != 2)
+  {
+    PCL_ERROR("Wrong input/output file count!\n");
+    displayHelp (argc, argv);
+    return (-1);
+  }
+
+  // Convert parsed output type to output type
+  int output_type (BINARY);
+  if (!parsed_output_type.empty ())
+  {
+    if (parsed_output_type == "ascii")
+      output_type = ASCII;
+    else if (parsed_output_type == "binary_compressed")
+      output_type = BINARY_COMPRESSED;
+    else
+    {
+      PCL_ERROR("Wrong output type!\n");
+      displayHelp (argc, argv);
+      return (-1);
+    }
+  }
+
+  // Find input/output file names
+  std::vector<int> file_args;
+  for (int i = 0; i < obj_files.size (); ++i)
+    file_args.push_back (obj_files[i]);
+  for (int i = 0; i < pcd_files.size (); ++i)
+    file_args.push_back (pcd_files[i]);
+  for (int i = 0; i < ply_files.size (); ++i)
+    file_args.push_back (ply_files[i]);
+  for (int i = 0; i < stl_files.size (); ++i)
+    file_args.push_back (stl_files[i]);
+  for (int i = 0; i < vtk_files.size (); ++i)
+    file_args.push_back (vtk_files[i]);
+  // Re-order, the first argument is always the input file
+  std::sort (file_args.begin (), file_args.end ());
+  if (file_args.size () != 2)
+  {
+    PCL_ERROR("Wrong input/output file count!\n");
+    displayHelp (argc, argv);
+    return (-1);
+  }
+
+  // If the input is a PCD file it can't be a mesh: treat it as a point cloud
+  if (boost::filesystem::path (argv[file_args[0]]).extension () == ".pcd")
+  {
+    pcl::PCLPointCloud2::Ptr cloud (new pcl::PCLPointCloud2);
+    //Eigen::Vector4f origin; // TODO: Support origin/orientation
+    //Eigen::Quaternionf orientation;
+    if (pcl::io::loadPCDFile (argv[file_args[0]], *cloud) < 0)  //, origin, orientation) < 0)
+    {
+      PCL_ERROR("Unable to load %s.\n", argv[file_args[0]]);
+      return (-1);
+    }
+
+    PCL_INFO("Loaded a point cloud with %d points (total size is %d) and the following channels:\n%s\n",
+             cloud->width * cloud->height, cloud->data.size (), pcl::getFieldsList (*cloud).c_str ());
+
+    if (!savePointCloud (cloud, argv[file_args[1]], output_type))
+    {
+      PCL_ERROR("Failed to save %s.\n", argv[file_args[0]]);
+      return (-1);
+    }
+  }
+
```

Comment:
Why are you writing a special case for pcd files and not following the same logic of trying a mesh first and then switching to point cloud?